interfaces/prompting/errors/errors.go

```diff
@@ -65,94 +54,109 @@ var (
 	ErrRuleExpirationInThePast = errors.New("cannot have expiration time in the past")
 )
 
+// Marker for UnsupportedValueError, should never be returned as an actual
+// error value.
+var ErrUnsupportedValue = errors.New("unsupported value")
+
+// UnsupportedValueError is a wrapper for errors about a field having an
+// unsupported value when there is a fixed set of supported values.
+type UnsupportedValueError struct {
+	Field     string
+	Msg       string
+	Value     interface{} // either string or []string
```

Comment:
```suggestion
	Value     any // either string or []string
```

Reply:
FWIW I also played with the following change locally:
```
 // UnsupportedValueError is a wrapper for errors about a field having an
 // unsupported value when there is a fixed set of supported values.
-type UnsupportedValueError struct {
+type UnsupportedValueError[V interface {
+       string | []string
+}] struct {
        Field     string
        Msg       string
-       Value     interface{} // either string or []string
+       Value     V
        Supported []string
 }
 
-func (e *UnsupportedValueError) Error() string {
+func (e *UnsupportedValueError[V]) Error() string {
        return e.Msg
 }
 
 // Unsupported value errors, which are built from the UnsupportedValueError struct
 
-func (e *UnsupportedValueError) Is(target error) bool {
+func (e *UnsupportedValueError[V]) Is(target error) bool {
        return target == ErrUnsupportedValue
 }
```
but I'm not sure it's worth it since it requires similar changes in api_prompting.go and some guesswork when converting to API errors